apis/database/v1alpha2/sql_types.go

```diff
@@ -212,13 +174,6 @@ type SQLServerStatus struct {
 
 	// Endpoint of the MySQL Server instance used in connection strings.
 	Endpoint string `json:"endpoint,omitempty"`
-
-	// RunningOperation stores any current long running operation for this
-	// instance across reconciliation attempts.
-	RunningOperation string `json:"runningOperation,omitempty"`
-
-	// RunningOperationType is the type of the currently running operation.
-	RunningOperationType string `json:"runningOperationType,omitempty"`
```

Comment:
Right, these are not exposed by the Azure API (as fields of the SQL server object). We used them to persist the running operation that was returned when we asked to create or delete a server. I'm guessing because it was the only obvious way to know that we'd asked for something to happen to the server given that AFAICT changes aren't reflected immediately in the API _and_ there's no way to query for running operations. (CC @jbw976 for history.)

We prefer to act based on real world observations rather than recorded status, so I switched to checking whether the name is taken to determine whether an SQL server is currently being created, which is the only operation we need to be aware of with the current implementation.

I agree that there will be more changes necessary to get this to v1beta1. My goal here was to unblock myself on the simple resource class selection front so I wanted to make minimal changes to the API and functionality. I believe @muvaf is going to be working on getting this to v1beta1 in the coming weeks, so hopefully this change moves the controller slightly in that direction.